app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -33,6 +33,38 @@ def delete_orphaned_repositories_for_mirror_proxies
           ::Katello::Pulp3::Repository.new(nil, smart_proxy).repositories_api.delete(repo._href)
         end
       end
+
+      def pulp3_enabled_repo_types(content_types = SETTINGS[:katello][:content_types].keys)
+        repo_types = content_types.collect do |content_type|
+          Katello::RepositoryTypeManager.find(content_type)
+        end
+
+        repo_types.select { |type| type.try(:pulp3_service_class) }
+      end
+
+      def delete_orphaned_distributions_for_mirror_proxies(repo_types = nil)
+        repo_types ||= pulp3_enabled_repo_types
+        repo_types.each do |repo_type|
+          pulp3_repo = repo_type.pulp3_service_class.new(nil, smart_proxy)
+          orphan_distribution_hrefs = ::Katello::Pulp3::Repository.orphan_distributions(pulp3_repo, {})
```

Comment:
This is somewhat abusing the pulp3 repository service class, which is maybe something we have to live with, but is there a way to not expose it to the caller of this orphan_distributions method?  My thinking, is if `orphan_distributions` just took a smart_proxy in as a parameter, and then internally created an instance (without a repository obj), then the caller wouldn't have to be aware of this.

Reply:
The issue that I've run into is that there is no global resource API for distributions. The distributions are provided by each plugin, so we'll need to work directly with each plugin's api for distributions and remotes.

Reply:
which i think i'm okay living with, but can we hide this detail from the caller of that method?

Reply:
If that's where we want to go, them we can push all of the repo type combing down into pulp3/repository.rb. That's the only way I can see how to keep that detail out of the smart proxy repository class.

Because identifying and then issuing deletes for distributions and remotes depend on the plugin apis, that means we have to use the appropriate binding class, right? I haven't tried to delete say, a file distribution through a Katello::Pulp3::Repository::AnsibleCollection distributions_api - would that even work?

There will be ugliness, I guess we have to decide where it lives.